api/tests/routes/pro/patch_venue_test.py

```diff
@@ -261,6 +276,86 @@ def test_update_venue_location_with_manual_edition(self, client, requests_mock)
             "old_info": "75000",
         }
 
+    @pytest.mark.parametrize("attribut_key,attribut_value", [("publicName", "Ma librairie"), ("name", "FNAC")])
+    def test_update_venue_public_name(self, attribut_key, attribut_value, client) -> None:
+        user_offerer = offerers_factories.UserOffererFactory()
+        venue = offerers_factories.VenueFactory(
+            name="old name",
+            managingOfferer=user_offerer.offerer,
+        )
+        offerer_address = venue.offererAddress
+        venue_siren = venue.managingOfferer.siren
+        venue_siret = venue.siret
+        venue_attr_value = getattr(venue, attribut_key)
+        venue_data = {
+            attribut_key: attribut_value,
+            "siret": venue_siren + "11111",
+        }
+        auth_request = client.with_session_auth(email=user_offerer.user.email)
+        response = auth_request.patch("/venues/%s" % venue.id, json=venue_data)
+
+        venue = offerers_models.Venue.query.one()
+        offerer_addresses = offerers_models.OffererAddress.query.order_by(offerers_models.OffererAddress.id).all()
+
+        assert response.status_code == 200
+        assert len(offerer_addresses) == 2
+        new_oa = offerer_addresses[-1]
+        assert getattr(venue, attribut_key) == venue_data.get(attribut_key)
+        assert venue.offererAddress.address.city == venue_data.get("city", venue.city)
+        assert venue.action_history[0].extraData["modified_info"] == {
+            attribut_key: {"new_info": attribut_value, "old_info": venue_attr_value},
+            "siret": {"new_info": venue_data["siret"], "old_info": venue_siret},
+            "offererAddress.address.banId": {
+                "new_info": "75101_9575_00003",
+                "old_info": "75102_7560_00001",
+            },
+            "offererAddress.address.inseeCode": {
+                "new_info": "75056",
+                "old_info": "75102",
+            },
+            "offererAddress.address.latitude": {
+                "new_info": "48.87171",
+                "old_info": "48.87004",
+            },
+            "offererAddress.address.longitude": {
+                "new_info": "2.308289",
+                "old_info": "2.3785",
+            },
+            "offererAddress.address.postalCode": {
+                "new_info": "75001",
+                "old_info": "75002",
+            },
+            "offererAddress.address.street": {
+                "new_info": "3 Rue de Valois",
+                "old_info": "1 boulevard Poissonnière",
+            },
+            "offererAddress.addressId": {"new_info": new_oa.addressId, "old_info": offerer_address.addressId},
+            "offererAddress.id": {
+                "new_info": new_oa.id,
+                "old_info": offerer_address.id,
+            },
+            "old_oa_label": {
+                "new_info": "old name",
+                "old_info": None,
+            },
+        }
+
+    def test_should_not_create_oa_at_update(self, client):
```

Comment:
Du détail, mais je trouve ça plus explicite sur ce qu’on cherche à tester:

```suggestion
    def test_should_not_create_oa_when_not_updating_location(self, client):
```